orchestrator-salt/src/main/resources/salt/salt/nginx/init.sls

```diff
@@ -1,3 +1,24 @@
+{% set gateway = salt['pillar.get']('gateway') %}
+
+/etc/nginx/sites-enabled/ssl.conf:
+  file.managed:
+    - makedirs: True
+    - source: salt://nginx/conf/ssl.conf
+
+# when gateway is defined, we do NOT config user facing cert and ssl for port 443, because services are available through knox on 8443
+# we still config ssl for port 9443 (for internal communication with ambari)
+{% if gateway.path is defined and gateway.path is not none %}
```

Comment:
Why we are checking gateway.path?

Reply:
Gateway object is always present in pillar, but the following properties of gateway are only set when the gatewaz is enabled/specified: 
```path, ssotype, ssoprovider, signpub, signcert, signkey, tokencert, topologies.``` 
Some of these can be nulls, but path is always set.